hyperspy/tests/io/test_ripple.py

```diff
@@ -0,0 +1,131 @@
+import tempfile
+import os.path
+import shutil
+
+import numpy as np
+import nose.tools as nt
+import numpy.testing as npt
+
+from hyperspy.io import load
+from hyperspy.signals import Signal, EELSSpectrum, EDSTEMSpectrum
+from hyperspy.io_plugins import ripple
+
+
+# Tuple of tuples (data shape, signal_dimensions)
+SHAPES_SDIM = (((3,), (1, )),
+               ((2, 3), (1, 2)),
+               ((2, 3, 4), (1, 2)),
+               )
+
+TMPDIR = tempfile.mkdtemp()
+MYPATH = os.path.dirname(__file__)
+
+nt.assert_equal.__self__.maxDiff = None
+
+
+def test_ripple():
+    try:
+        for dtype in ripple.dtype2keys.keys():
+            for shape, dims in SHAPES_SDIM:
+                for dim in dims:
+                    yield _run_test, dtype, shape, dim
+    finally:
+        shutil.rmtree(TMPDIR)
```

Comment:
Wouldn't `TMPDIR.cleanup()` be better, [as per the docs](https://docs.python.org/3.5/library/tempfile.html#tempfile.TemporaryDirectory) ?